src/app/store/test-records/effects/test-records.effects.ts

```diff
@@ -191,6 +198,22 @@ export class TestResultsEffects {
         mergedForms.testTypes[0].testTypeId = id;
         mergedForms.testTypes[0].name = testTypeTaxonomy?.name ?? '';
         mergedForms.testTypes[0].testTypeName = testTypeTaxonomy?.testTypeName ?? '';
+        mergedForms.typeOfTest = (testTypeTaxonomy?.typeOfTest as TypeOfTest) ?? TypeOfTest.CONTINGENCY;
+
+        const now = new Date().toISOString();
+
+        if (mergedForms.typeOfTest !== TypeOfTest.CONTINGENCY) {
+          mergedForms.testerName = user.name;
+          mergedForms.testerEmailAddress = user.username;
+          mergedForms.testerStaffId = user.oid;
+          mergedForms.testStartTimestamp = now;
+          mergedForms.testEndTimestamp = now;
+          mergedForms.testTypes[0].testTypeStartTimestamp = now;
+          mergedForms.testTypes[0].testTypeEndTimestamp = now;
```

Comment:
I don't know if it matters (the API might do it for use anyways) but we might want to set these in the submission effect rather than on initialisation, depending on if we want submission time or completion time, or set the start time here and the end one on the submission one, probably something to discuss with BA/PO

Reply:
yeah bus for the sake of a couple of minutes I hope it doesn't matter. Rather not change the submission effect as well.